libhb/handbrake/qsv_common.h

```diff
@@ -226,7 +226,8 @@ typedef struct QSVFrame {
     struct QSVFrame *next;
 } QSVFrame;
 
-#define HB_POOL_SURFACE_SIZE (64)
+// Large size is driven by hevc which has a maximum GOP size of 256
+#define HB_POOL_SURFACE_SIZE (268)
```

Comment:
As I remember HB_POOL_SURFACE_SIZE will be used as ArraySize parameter in D3D11_TEXTURE2D_DESC.
https://docs.microsoft.com/en-us/windows/win32/api/d3d11/ns-d3d11-d3d11_texture2d_desc

I am not sure that it is possible to create so large array in the texture, that is why two pools were created in EncQSVFramesContext.. but let me check

Reply:
Thanks .. DirectX isn't something I have much experience with .. and the pool size is fine under linux so that is a potential problem that could easily have escaped my cursory windows testing. It did however work for me under windows with a post patch HandbrakeCLI.  As the docs you link claim the max array size is 2048 I was thinking that if it's not because the problem is fixed elsewhere it might have something to do with out of internal graphics memory type errors. I think it's only intel chips with Iris graphics which have a small dedicated video memory .. for all HD graphics chips (like my skylake) it's all system memory so won't run out.

I think the use of internal graphics memory is decided here as the called function reaches into the windows specific code.

https://github.com/HandBrake/HandBrake/blob/b34a00ee0e33e2400be889cf20aa6643ead5a6be/libhb/enc_qsv.c#L1036

Limits to what I can do without installing windows. If you get back to me saying you've got a concern then I promise to do it and spend some time figuring out the frame pooling code.

Reply:
we need to come up yet with proper solution about amount of resources allocated and really needed.

ffmpeg does require additional param for the cases when bigger(>64) pool is really needed: https://github.com/FFmpeg/FFmpeg/blob/6434b84460bb8b996bc52b33b2012e6b05d3a459/fftools/ffmpeg_qsv.c#L96

Reply:
It does not allow to create so large d3d11 texture underneath in FFMPEG. I tested it with latest Windows driver on Intel NUC. 64 is a maximum number.

Reply:
@agalin89 Have submitted a very simple test fix for this. As I can't reproduce the errors you were getting could you please test if for me. See the comment:

https://github.com/HandBrake/HandBrake/pull/2386#issuecomment-564429835

Thanks!